### ISSUE TITLE
Display row counts, table size, etc in output messages

### ISSUE THREAD

**Issue by chdsbd** [sbdchd/squawk#197]
Related to #116

We could accept an input file that would include table name, row count, index size, table size, etc. and use that information to display more useful error messages.

If a migration is going to lock a table while it scans the whole table, that doesn't matter for a 300 row table, but does for a 3,000,000 row table.

**Comment by jelder** [comment]
Are rules that require access to a database welcome in squawk?

That said, the database instance which migrations are developed against is usually not biggest database that it will be run against. And locking a million-row table that gets no queries may be ok, but locking a thousand-row table that gets queried constantly could be bad.

Printing warnings based on the size of tables in a development environment may be misleading.

**Comment by chdsbd** [comment]
@jelder We don't have any rules that connect to the database, but I'm definitely open to ideas for rules that require more database information

I think my original idea for this issue was that you'd run some SQL on your production database, then Squawk would use that information for linting

**Comment by jelder** [comment]
@chdsbd One thing I'd like to have is a warning that a foreign key constraint isn't indexed. Static analysis of the migration file can't provide enough information to eliminate false positives for such a rule.

Such a query might look like this: 

```sql
WITH indexes AS (
SELECT
    n.nspname as schema_name,
    t.relname as table_name,
    a.attname as column_name
FROM
    pg_class t
JOIN
    pg_index i ON t.oid = i.indrelid
JOIN
    pg_attribute a ON a.attrelid = t.oid AND a.attnum = ANY(i.indkey)
JOIN
    pg_namespace n ON t.relnamespace = n.oid
WHERE
    t.relkind = 'r' -- real tables
    AND array_position(i.indkey, a.attnum) = 0 -- only first column of possibly compound index
),
foreign_keys AS (
SELECT
    n.nspname AS schema_name,
    cl.relname AS table_name,
    a.attname AS column_name,
    ct.conname AS constraint_name
FROM
    pg_constraint ct
    JOIN pg_class cl ON ct.conrelid = cl.oid
    JOIN pg_namespace n ON cl.relnamespace = n.oid
    JOIN pg_attribute a ON a.attnum = ANY(ct.conkey) AND a.attrelid = cl.oid
WHERE
    ct.contype = 'f'
)
SELECT
    schema_name as "schema!",
    table_name as "table!",
    column_name as "name!"
FROM foreign_keys
LEFT JOIN indexes USING (schema_name, table_name, column_name)
WHERE indexes.column_name IS NULL
AND schema_name = ANY($1)
```

(query is not perfect but you get the idea)
